common/model/tests/test_bundle.c

```diff
@@ -33,10 +33,78 @@ void test_normalized_bundle(void) {
   TEST_ASSERT_EQUAL_MEMORY(bytes, normalized_bundle_bytes, length);
 }
 
+void test_bundle_finalize(void) {
+  bundle_transactions_t *bundle = NULL;
+  flex_trit_t addr_sender[FLEX_TRIT_SIZE_243] = {};
+  flex_trit_t addr_reciver[FLEX_TRIT_SIZE_243] = {};
+  flex_trit_t tag[FLEX_TRIT_SIZE_81] = {};
+  flex_trit_t obsolete_tag[FLEX_TRIT_SIZE_81] = {};
+  iota_transaction_t tx;
+  iota_transaction_t *bundle_tx;
+  transaction_reset(&tx);
+  flex_trits_from_trytes(addr_reciver, NUM_TRITS_ADDRESS,
+    (const tryte_t *)"LQZYGHAQLJLENO9IBSFOFIYHIBKOHEWVAEHKYOED9WBERCCLGGLOJVIZSIUUXGJ9WONIGBXKTVAWUXNHW",
+    NUM_TRYTES_ADDRESS, NUM_TRYTES_ADDRESS);
+  flex_trits_from_trytes(addr_sender, NUM_TRITS_ADDRESS,
+    (const tryte_t *)"SEPKTEEOIIVHFJNQRHX9VXKQDYTXXRTF9TYXBIBXMYEXFDUKOTLNGCMRIWXMNT9ZBJUCXFWTNHGPAZESX",
+    NUM_TRYTES_ADDRESS, NUM_TRYTES_ADDRESS);
+  flex_trits_from_trytes(tag, NUM_TRITS_TAG,
+                         (const tryte_t *)"IOTA99999999999999999999999",
+                         NUM_TRYTES_TAG, NUM_TRYTES_TAG);
+  flex_trits_from_trytes(obsolete_tag, NUM_TRITS_TAG,
+                         (const tryte_t *)"9PTA99999999999999999999999",
+                         NUM_TRYTES_TAG, NUM_TRYTES_TAG);
+
+  bundle_transactions_new(&bundle);
+  // reciver tx
+  transaction_set_address(&tx, addr_reciver);
+  transaction_set_tag(&tx, tag);
+  transaction_set_obsolete_tag(&tx, tag);
+  transaction_set_timestamp(&tx, 1546760794);
+  transaction_set_value(&tx, 1000);
+  transaction_set_current_index(&tx, 0);
+  transaction_set_last_index(&tx, 2);
+  tx.loaded_columns_mask |= MASK_ALL_COLUMNS;
+  bundle_transactions_add(bundle, &tx);
+
+  // sender tx
+  transaction_reset(&tx);
+  transaction_set_address(&tx, addr_sender);
+  transaction_set_tag(&tx, tag);
+  transaction_set_obsolete_tag(&tx, tag);
+  transaction_set_timestamp(&tx, 1546760811);
+  transaction_set_value(&tx, -1000);
+  transaction_set_current_index(&tx, 1);
+  transaction_set_last_index(&tx, 2);
+  tx.loaded_columns_mask |= MASK_ALL_COLUMNS;
+  bundle_transactions_add(bundle, &tx);
+  // security level 2
+  transaction_reset(&tx);
+  transaction_set_address(&tx, addr_sender);
+  transaction_set_tag(&tx, tag);
+  transaction_set_obsolete_tag(&tx, tag);
+  transaction_set_timestamp(&tx, 1546760811);
+  transaction_set_value(&tx, 0);
+  transaction_set_current_index(&tx, 2);
+  transaction_set_last_index(&tx, 2);
+  tx.loaded_columns_mask |= MASK_ALL_COLUMNS;
+  bundle_transactions_add(bundle, &tx);
+
+  bundle_finalize(bundle);
+  bundle_dump(bundle);
```

Comment:
Remove

Reply:
why?

Reply:
Assertions are sufficient, IMHO we shouldn't pollute stdout with logs if we are looking for information...

Reply:
Whoops I overlooked, you're right